tools/provision.sh

```diff
@@ -2,13 +2,25 @@
 set -euxo pipefail
 
 source /etc/os-release
+ARCH=$(uname -m)
 
 # koji and ansible are not in RHEL repositories. Depending on them in the spec
 # file breaks RHEL gating (see OSCI-1541). Therefore, we need to enable epel
 # and install koji and ansible here.
-if [[ $ID == rhel || $ID == centos ]]; then
-    sudo dnf install -y https://dl.fedoraproject.org/pub/epel/epel-release-latest-8.noarch.rpm
+if [[ $ID == rhel || $ID == centos ]] && [[ ${VERSION_ID%.*} == 8 ]] && ! rpm -q epel-release; then
+    curl -Ls --retry 5 --output /tmp/epel.rpm \
+        https://dl.fedoraproject.org/pub/epel/epel-release-latest-8.noarch.rpm
+    sudo rpm -Uvh /tmp/epel.rpm
     sudo dnf install -y koji ansible
+elif [[ $ID == rhel || $ID == centos ]] && [[ ${VERSION_ID%.*} == 9 ]]; then
+    # we have our own small epel for EL9, let's install it
+
+    # install Red Hat certificate, otherwise dnf copr fails
+    curl -LO --insecure https://hdn.corp.redhat.com/rhel8-csb/RPMS/noarch/redhat-internal-cert-install-0.1-23.el7.csb.noarch.rpm
+    sudo dnf install -y ./redhat-internal-cert-install-0.1-23.el7.csb.noarch.rpm dnf-plugins-core
+    sudo dnf copr enable -y copr.devel.redhat.com/osbuild-team/epel-el9 "rhel-9.dev-$ARCH"
```

Comment:
Could you please explain why it is needed to pass `"rhel-9.dev-$ARCH"` to the `dnf copr` command? I would expect everything to work just fine even without it.

Reply:
I don't have an explanation I just took it from this commit from Ondrej https://github.com/osbuild/osbuild-composer/commit/e1462e261db3583628c73d92705965101e000842

Reply:
hmm, not sure why Ondrej used it. I never saw it being used, that't the reason why I'm asking... I think it should not be needed, but can't hurt to keep it there if it works. My motivation was only that we would not need to define `ARCH` variable in that case 😄

Reply:
I can try removing it and see what happens :D ... so it would be just `"rhel-9.dev"` ?

Reply:
@jrusz no, please try just `sudo dnf copr enable -y copr.devel.redhat.com/osbuild-team/epel-el9`

Reply:
Alright thanks. The base tests seemed to have passed, I just added this change and if everything passes I think we could be ready to merge this :)

Reply:
I just tested the `dnf copr` command as I suggested and for some reason it does not work with that copr repo if the "chroot" is not specified. I have no idea why, as I never saw it being used, but I think that we need to revert back to the original stat. Sorry for the noise 😉

Reply:
Np, I reverted it. Let's see if the CI passes now with all the tests enabled :)